bson-kotlinx/src/test/kotlin/org/bson/codecs/kotlinx/KotlinSerializerCodecTest.kt

```diff
@@ -472,6 +473,52 @@ class KotlinSerializerCodecTest {
         assertRoundTrips(expected, dataClass)
     }
 
+    @Test
+    fun testDataClassWithOptionalObjectIdAndBsonDocument() {
+        val subDocument =
+            """{
+    | "_id": 1,
```

Comment:
It's not clear to me why it's necessary to test with such a complex subdocument.  What's being tested by this?  The fact that I don't know makes me think I don't understand the issue.

Reply:
The subdocument is just copy-pasta.

I'll update the tests and move to a const.